### PR TITLE
HBASE-26108 add option to disable scanMetrics in TableSnapshotInputFo…

### PR DESCRIPTION
**Pull request by huaxiangsun**
…rmat

### CODE REVIEW

File: hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/TableSnapshotInputFormatImpl.java [hunk 1]
```diff
@@ -107,6 +107,14 @@
   public static final String SNAPSHOT_INPUTFORMAT_ROW_LIMIT_PER_INPUTSPLIT =
       "hbase.TableSnapshotInputFormat.row.limit.per.inputsplit";
 
+  /**
+   * Whether to enable scan metrics on Scan, default to true
+   */
+  public static final String  SNAPSHOT_INPUTFORMAT_SCAN_METRICS_ENABLED =
+    "hbase.TableSnapshotInputFormat.scan_metrics_enabled";
```
Comment (anoopsjohn) [inline hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/TableSnapshotInputFormatImpl.java:114]: Should we name it as scan_metrics.enabled ?  That is what normally we do I believe and I could see many such conf names in code base

File: hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/TableSnapshotInputFormatImpl.java [hunk 2]
```diff
@@ -381,7 +388,11 @@ public static Scan extractScanFromConf(Configuration conf) throws IOException {
     boolean localityEnabled = conf.getBoolean(SNAPSHOT_INPUTFORMAT_LOCALITY_ENABLED_KEY,
                                               SNAPSHOT_INPUTFORMAT_LOCALITY_ENABLED_DEFAULT);
 
-    List<InputSplit> splits = new ArrayList<>();
+    boolean scanMetricsEnabled = conf.getBoolean(SNAPSHOT_INPUTFORMAT_SCAN_METRICS_ENABLED,
+      SNAPSHOT_INPUTFORMAT_SCAN_METRICS_ENABLED_DEFAULT);
+    scan.setScanMetricsEnabled(scanMetricsEnabled);
+
+      List<InputSplit> splits = new ArrayList<>();
```
Comment (anoopsjohn) [inline hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/TableSnapshotInputFormatImpl.java:395]: nit :  Not proper indentation

**Review by anoopsjohn — APPROVED** [review]

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  1s |  master passed  |
| +1 :green_heart: |  compile  |   0m 26s |  master passed  |
| +1 :green_heart: |  shadedjars  |   8m 16s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 40s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m  6s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 19s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  10m 43s |  hbase-mapreduce in the patch passed.  |
|  |   |  38m  3s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3516 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 9bef1ff822b2 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 21c4578f22 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/1/testReport/ |
| Max. process+thread count | 4524 (vs. ulimit of 30000) |
| modules | C: hbase-mapreduce U: hbase-mapreduce |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 58s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 20s |  master passed  |
| +1 :green_heart: |  compile  |   0m 48s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 18s |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 45s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 45s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 18s |  hbase-mapreduce: The patch generated 1 new + 1 unchanged - 0 fixed = 2 total (was 1)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  20m  4s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  spotbugs  |   0m 53s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  41m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3516 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 39326a6663c8 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 21c4578f22 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-mapreduce.txt |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-mapreduce U: hbase-mapreduce |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 26s |  master passed  |
| +1 :green_heart: |  compile  |   0m 33s |  master passed  |
| +1 :green_heart: |  shadedjars  |   9m 31s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 25s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  9s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   9m 26s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 23s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  16m 15s |  hbase-mapreduce in the patch passed.  |
|  |   |  50m  7s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3516 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux b12efc853cd6 4.15.0-142-generic #146-Ubuntu SMP Tue Apr 13 01:11:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 21c4578f22 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/1/testReport/ |
| Max. process+thread count | 2888 (vs. ulimit of 30000) |
| modules | C: hbase-mapreduce U: hbase-mapreduce |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/TableSnapshotInputFormatImpl.java (hunk 1 above)
Comment (huaxiangsun) [reply hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/TableSnapshotInputFormatImpl.java:114]: Thanks for the review, let me change it.

File: hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/TableSnapshotInputFormatImpl.java (hunk 2 above)
Comment (huaxiangsun) [reply hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/TableSnapshotInputFormatImpl.java:395]: Will push another PR.

**Comment by huaxiangsun** [comment]
Update to address Anoop's comments.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 27s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  1s |  master passed  |
| +1 :green_heart: |  compile  |   0m 25s |  master passed  |
| +1 :green_heart: |  shadedjars  |   8m  7s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 25s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 17s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 19s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  10m 28s |  hbase-mapreduce in the patch passed.  |
|  |   |  37m 50s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3516 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux cca88e0ce24f 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / be2c97e203 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/2/testReport/ |
| Max. process+thread count | 4402 (vs. ulimit of 30000) |
| modules | C: hbase-mapreduce U: hbase-mapreduce |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 20s |  master passed  |
| +1 :green_heart: |  compile  |   1m  1s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m  5s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 20s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  21m 49s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  spotbugs  |   0m 56s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 13s |  The patch does not generate ASF License warnings.  |
|  |   |  47m 15s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3516 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 4d4e34e1a814 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / be2c97e203 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-mapreduce U: hbase-mapreduce |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  5s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  7s |  master passed  |
| +1 :green_heart: |  compile  |   0m 33s |  master passed  |
| +1 :green_heart: |  shadedjars  |   9m 26s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 23s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 32s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   9m 29s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 23s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  15m 22s |  hbase-mapreduce in the patch passed.  |
|  |   |  49m 14s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3516 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 7b009ccacc7a 4.15.0-142-generic #146-Ubuntu SMP Tue Apr 13 01:11:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / be2c97e203 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/2/testReport/ |
| Max. process+thread count | 2910 (vs. ulimit of 30000) |
| modules | C: hbase-mapreduce U: hbase-mapreduce |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3516/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**